src/core/batch.rkt

```diff
@@ -82,71 +82,173 @@
       (unmunge root)))
   exprs)
 
-(define (expand-taylor batch)
-  (define vars (batch-vars batch))
+(define (expand-taylor input-batch)
```

Comment:
`expand-taylor` batch is implemented sequentially rather than relying on the tree of expression.
The code is tested and seems to be correct (no changes on nightly previously as well).